src/core/link-to-dfn.js

```diff
@@ -169,8 +169,12 @@ function findMatchingDfn(anchor, titleToDfns) {
 function processAnchor(anchor, dfn, titleToDfns) {
   let noLocalMatch = false;
   const { linkFor } = anchor.dataset;
+  const { dfnFor } = dfn.dataset;
   if (dfn.dataset.cite) {
     anchor.dataset.cite = dfn.dataset.cite;
+  } else if (linkFor === dfnFor) {
```

Comment:
This happens when the "dfnFor" is defined externally, so `titleToDfns` doesn't have dfnFor/linkFor.